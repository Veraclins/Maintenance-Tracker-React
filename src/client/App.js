import React from 'react';
import PropTypes from 'prop-types';
import { ConnectedRouter } from 'connected-react-router';
import './styles/styles.css';
import routes from './routes/routes';

export const App = ({ history }) => (
  <ConnectedRouter history={history}>
    { routes }
  </ConnectedRouter>
);

App.propTypes = {
  history: PropTypes.shape({}).isRequired,
};

export default App;
