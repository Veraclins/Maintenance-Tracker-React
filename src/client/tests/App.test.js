import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import App from '../App';

const history = createBrowserHistory();


it('renders without crashing', () => {
  const wrapper = shallow(<App history={history} />);
  expect(wrapper.find('div'));
  expect(wrapper.find('NavBar'));
  expect(wrapper.find('Switch'));
  expect(wrapper).toMatchSnapshot();
});
