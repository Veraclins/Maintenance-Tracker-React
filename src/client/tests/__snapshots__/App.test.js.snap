// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Connect(ConnectedRouter)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <NavBar />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/login"
      />
    </Switch>
  </div>
</Connect(ConnectedRouter)>
`;
